flocker/node/test/test_change.py

```diff
@@ -521,3 +523,47 @@ def test_failed(self, logger):
         action._logger = logger
         failure = self.failureResultOf(run_state_change(action, DEPLOYER))
         self.assertEqual(failure.getErrorMessage(), "Oh no")
+
+    def assert_nested_logging(self, combo, action_type, logger):
+        """
+        All the underlying ``IStateChange`` will be run in Eliot context in
+        which the sequential ``IStateChange`` is run, even if they are not
+        run immediately.
+
+        :param combo: ``sequentially`` or ``in_parallel``.
+        :param action_type: ``eliot.ActionType`` we expect to be parent of
+            sub-changes logs.
+        :param logger: A ``MemoryLogger`` where messages go.
+        """
+        actions = [ControllableAction(result=Deferred()),
+                   ControllableAction(result=succeed(None))]
+        for action in actions:
+            self.patch(action, "_logger", logger)
```

Comment:
Why `patch` rather than straight-up assignment? Also, why not make `logger` an optional constructor parameter for `ControllableAction`?